### PR TITLE
Moved some things outside loops, project-wide.

### PR DESCRIPTION
**Pull request by frankmayer**
### Summary of Changes
- Moved some things outside loops.

### Testing Instructions

None, should not change behavior

### Documentation Changes Required

None.

### CODE REVIEW

**Review by ggppdk — APPROVED** [review]

File: libraries/legacy/controller/form.php [hunk 1]
```diff
@@ -255,18 +255,18 @@ public function batch($model)
 		// Build an array of item contexts to check
 		$contexts = array();
 
+		if (isset($this->extension))
+		{
+			$option = $this->extension;
+		}
+		else
+		{
+			$option = $this->option;
```
Comment (zero-24) [inline libraries/legacy/controller/form.php]: Can't we remove the else also? So we defaulting to $this->option and only if the extension is set we set that?

Comment (frankmayer) [inline libraries/legacy/controller/form.php]: In some other cases I would do that, too, but in this case I would prefer to leave it like this, as
- it is only a small if/else that is only setting the `$option` without introducing further code in the groups.
- it is still easy to read 
- we don't need to assign `$option` with a value twice in the case of an extension.
- there is no theoretical default (at least for me) in this function. as to the value of `$option`

Comment (frankmayer) [inline libraries/legacy/controller/form.php]: No, sometimes we just don't see the tree... 
You're right.

Comment (dgt41) [inline libraries/legacy/controller/form.php]: @frankmayer you can drop the else there"
```php
$option = $this->option;

if (isset($this->extension))
{
 $option = $this->extension;
}
```

Comment (frankmayer) [inline libraries/legacy/controller/form.php]: @dgt41 BTW, your example would mean that if it is an extension, we would have assigned something to $option twice. With the if-else or the ternary it's only one assignment in either case.

Comment (frankmayer) [inline libraries/legacy/controller/form.php]: Note, there has been a correction to the numbers and script, above.

Comment (dgt41) [inline libraries/legacy/controller/form.php]: @frankmayer thanks! I really thought it was the other way around 😕

Comment (frankmayer) [inline libraries/legacy/controller/form.php]: You're welcome!! 
Here are some numbers with another method added. Just assigning something to `$options` twice, without any comparison:

```
Took 0.814069 seconds for ternary
Took 0.829895 seconds for ifElse
Took 0.873709 seconds for reAssign
Took 0.788505 seconds for assignVariableTwice
```
BTW, I was doing those with PHP 7.1 . There might be completely different numbers with 5.6 and even with 7.0. I'll check those, too, out of interest.

Comment (frankmayer) [inline libraries/legacy/controller/form.php]: @dgt41 

PHP 5.6 (with xdebug enabled)
```
Took 0.976332 seconds for ternary
Took 0.982323 seconds for ifElse
Took 1.080398 seconds for reAssign
Took 0.883778 seconds for assignVariableTwice
```

PHP 7.0 (without xdebug disabled)
```
Took 0.129902 seconds for ternary
Took 0.129425 seconds for ifElse
Took 0.142026 seconds for reAssign
Took 0.137943 seconds for assignVariableTwice
```

BTW, xdebug was enabled in php.ini in the test results with PHP 7.1, that's why PHP 5.6 and 7.1' times are much higher than the times of PHP 7.0.

**Review by andrepereiradasilva — APPROVED** [review]

File: administrator/components/com_media/models/list.php [hunk 2]
```diff
@@ -209,9 +209,11 @@ public function getList()
 		// Iterate over the folders if they exist
 		if ($folderList !== false)
 		{
+			$tmpBaseObject = new JObject;
+
 			foreach ($folderList as $folder)
 			{
-				$tmp = new JObject;
+				$tmp = clone $tmpBaseObject;
```
Comment (laoneo) [inline administrator/components/com_media/models/list.php]: Does that improve anything? did you do that because of performance?

Comment (frankmayer) [inline administrator/components/com_media/models/list.php]: Yes, cloning an object is cheaper than instantiating, when in loops, so I instantiate a base object and clone that inside the loop.

Comment (laoneo) [inline administrator/components/com_media/models/list.php]: According to this thread http://stackoverflow.com/questions/18040137/which-is-more-efficient-when-creating-many-objects-together-in-php the gain is minimal. I would leave it as it is. Looks for me like over engineering stuff. As it is now, the code looks more clear on the first sight.

**Review by andrepereiradasilva — APPROVED** [review]

**Review by ggppdk — APPROVED** [review]

**Review by izharaazmi — APPROVED** [review]

File: administrator/components/com_media/models/list.php [hunk 3]
```diff
@@ -209,9 +209,11 @@ public function getList()
 		// Iterate over the folders if they exist
 		if ($folderList !== false)
 		{
+			$tmpBaseObject = new JObject;
```
Comment (frankmayer) [inline administrator/components/com_media/models/list.php]: I have explained it a few lines above, even with benchmarks...
see: https://github.com/joomla/joomla-cms/pull/13249/files/cb97b400303dea8cc849848786896982a2161745#diff-085bd8a62e18ac0bff4afba700ae9863R216

File: libraries/cms/html/menu.php [hunk 4]
```diff
@@ -328,21 +328,21 @@ public static function treerecurse($id, $indent, $list, &$children, $maxlevel =
...
+			{
+				$pre = '<sup>|_</sup>&#160;';
+				$spacer = '.&#160;&#160;&#160;&#160;&#160;&#160;';
+			}
+			else
+			{
+				$pre = '- ';
+				$spacer = '&#160;&#160;';
+			}
+
```
Comment (frankmayer) [inline libraries/cms/html/menu.php]: Hmm, not sure if it is in the Joomla code-style, but I just moved it out as it was, without changing CS. Will look at it though.

Comment (andrepereiradasilva) [inline libraries/cms/html/menu.php]: more or less see http://joomla.github.io/coding-standards/?coding-standards/chapters/php.md (Function calls block)

File: administrator/components/com_media/models/list.php (hunk 2 above)
Comment (frankmayer) [inline administrator/components/com_media/models/list.php]: @laoneo I'd like to leave this change though if it's not a too big thing to resolve for you, if needed.

**Review by andrepereiradasilva — APPROVED** [review]

File: plugins/system/fields/fields.php [hunk 5]
```diff
@@ -145,6 +145,13 @@ public function onContentAfterSave($context, $item, $isNew)
 		// Loading the model
 		$model = JModelLegacy::getInstance('Field', 'FieldsModel', array('ignore_request' => true));
 
+		$id = null;
```
Comment (frankmayer) [inline plugins/system/fields/fields.php]: Reverted changes in this file to not clash with @laoneo's refactoring efforts

**Comment by frankmayer** [comment]
Any specific reason, this has not been merged yet? 
The code-changes in question were approved my multiple reviewers.

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on dc694791b38949e154ffa5521150bb381e7e5fb0<br /><br />On code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13249">issues.joomla.org/tracker/joomla-cms/13249</a>.</sub>

**Comment by frankmayer** [comment]
One code review test and multiple approvals. Can we please RTC this? Thank you

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on dc694791b38949e154ffa5521150bb381e7e5fb0<br /><br />Code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13249">issues.joomla.org/tracker/joomla-cms/13249</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Pull request merged**